libglusterfs/src/dict.c

```diff
@@ -1468,7 +1468,31 @@ dict_reset(dict_t *dict)
                          "dict is NULL");
         goto out;
     }
-    dict_foreach(dict, dict_remove_foreach_fn, NULL);
+
+    LOCK(&dict->lock);
+    data_pair_t *curr = dict->members_list;
+    data_pair_t *next = NULL;
+
+    while (curr != NULL) {
```

Comment:
The code is essentially like dict_destroy(), just without the deletion of the dictionary, no? I wonder if we can reuse some code. Otherwise, looks OK.

Reply:
correct, i'll place the iteration block in a separate function for reusing.